dali/kernels/CMakeLists.txt

```diff
@@ -26,4 +26,4 @@ collect_test_sources(DALI_KERNEL_TEST_SRCS)
 cuda_add_library(${dali_kernel_lib} STATIC "${DALI_KERNEL_SRCS}")
 cuda_add_library(${dali_kernel_test_lib} STATIC "${DALI_KERNEL_TEST_SRCS}")
 
-target_link_libraries(${dali_kernel_test_lib} ${dali_kernel_lib})
+target_link_libraries(${dali_kernel_test_lib} PRIVATE ${dali_kernel_lib})
```

Comment:
Now you force the test binary to link `${dali_kernel_lib}` explicitly. I'd say it's `PUBLIC`.